src/psyclone/domain/lfric/kernel_arg_order.py

```diff
@@ -0,0 +1,1063 @@
+        2D integer array with intent in and kind i_def. The size of
+        the dimensions are ndf_<function_space>, 2. Its default name
+        is "boundary_dofs_"<field_name>, where <field_name> is the
+        default name of the field on the current function space.
+
+        :raises InternalError: if the enforce_bc_kernel does not have \
+            a single field argument on the any_space_1 function space.
+
+        '''
+        # Check that this kernel has a single field argument that is
+        # on the any_space_1 function space.
+        if len(self._metadata.meta_args) != 1:
+            raise InternalError(
+                f"An enforce_bc_code kernel should have a single "
+                f"argument but found '{len(self._metadata.meta_args)}'.")
+        meta_arg = self._metadata.meta_args[0]
+        if not type(meta_arg) == FieldArgMetadata:
+            raise InternalError(
+                f"An enforce_bc_code kernel should have a single field "
+                f"argument but found '{type(meta_arg).__name__}'.")
+        if not meta_arg.function_space == "any_space_1":
+            raise InternalError(
+                f"An enforce_bc_code kernel should have a single field "
+                f"argument on the 'any_space_1' function space, but found "
+                f"'{meta_arg.function_space}'.")
+
+        field_name = self._field_name(meta_arg)
+        self.arg_info.append(f"boundary_dofs_{field_name}")
+        self._arg_index += 1
+
+    def operator_bcs_kernel(self):
+        '''Fix for the operator boundary condition kernel. Adds a boundary
+        dofs 2D integer array with intent in and kind i_def. The size
+        of the dimensions are ndf_<function_space>, 2. Its default
+        name is "boundary_dofs_"<field_name>, where <field_name> is
+        the default name of the field on the current function space.
+
+        :raises InternalError: if the enforce_operator_bc_kernel does \
+            not have a single lma operator argument.
+
+        '''
+        # Check that this kernel has a single LMA argument.
+        if len(self._metadata.meta_args) != 1:
+            raise InternalError(
+                f"An enforce_operator_bc_code kernel should have a single "
+                f"argument but found '{len(self._metadata.meta_args)}'.")
+        meta_arg = self._metadata.meta_args[0]
+        if not type(meta_arg) == OperatorArgMetadata:
+            raise InternalError(
+                f"An enforce_operator_bc_code kernel should have a single "
+                f"lma operator argument but found "
+                f"'{type(meta_arg).__name__}'.")
+
+        lma_operator_name = self._operator_name(meta_arg)
+        self.arg_info.append(f"boundary_dofs_{lma_operator_name}")
+        self._arg_index += 1
+
+    def stencil_2d_unknown_extent(self, meta_arg):
+        '''The field entry has a stencil access of type cross2d so add a 1D
```

Comment:
"...so add a stencil-size argument which is an integer array...."

Reply:
Not required in this PR.